pkg/k8scontext/context.go

```diff
@@ -352,7 +357,63 @@ func (c *Context) GetGateways() []*v1alpha3.Gateway {
 	return annotatedGateways
 }
 
-func isIngressApplicationGateway(ingress *v1beta1.Ingress) bool {
+// AddIngressStatus adds IP address in Ingress Status
+func (c *Context) AddIngressStatus(ingress v1beta1.Ingress, address IPAddress) error {
+	loadBalancerIngresses := ingress.Status.LoadBalancer.Ingress
+
+	// Skip if already added.
+	for _, loadBalancerIngress := range loadBalancerIngresses {
+		if loadBalancerIngress.IP == string(address) {
+			return nil
+		}
+	}
+
+	loadBalancerIngresses = append(loadBalancerIngresses, v1.LoadBalancerIngress{
+		IP: string(address),
+	})
+
+	ingress.Status.LoadBalancer.Ingress = loadBalancerIngresses
+
+	if _, err := c.kubeClient.ExtensionsV1beta1().Ingresses(ingress.Namespace).UpdateStatus(&ingress); err != nil {
+		errorLine := fmt.Sprintf("Unable to update ingress %s/%s status: error %s", ingress.Namespace, ingress.Name, err.Error())
+		glog.Error(errorLine)
+		return errors.New(errorLine)
+	}
+
+	return nil
+}
+
+// RemoveIngressStatus removes IP address in Ingress Status
+func (c *Context) RemoveIngressStatus(ingress v1beta1.Ingress, address IPAddress) error {
+	loadBalancerIngresses := ingress.Status.LoadBalancer.Ingress
+
+	// find the status the needs to be removed
+	removeIdx := -1
+	for idx, loadBalancerIngress := range loadBalancerIngresses {
+		if loadBalancerIngress.IP == string(address) {
+			removeIdx = idx
+			break
+		}
+	}
+
+	if removeIdx != -1 {
+		loadBalancerIngresses[removeIdx] = loadBalancerIngresses[len(loadBalancerIngresses)-1]
+		loadBalancerIngresses = loadBalancerIngresses[:len(loadBalancerIngresses)-1]
+	}
+
+	ingress.Status.LoadBalancer.Ingress = loadBalancerIngresses
+
+	if _, err := c.kubeClient.ExtensionsV1beta1().Ingresses(ingress.Namespace).UpdateStatus(&ingress); err != nil {
+		errorLine := fmt.Sprintf("Unable to update ingress %s/%s status: error %s", ingress.Namespace, ingress.Name, err.Error())
+		glog.Error(errorLine)
+		return errors.New(errorLine)
```

Comment:
I am returning this as I think it would be mostly 403 which might not be sufficient.